### PR TITLE
USHIFT-1265: update configure-vm.sh to install go 1.20.4

### PR DESCRIPTION
**Pull request by dhellmann**

### CODE REVIEW

**Comment by dhellmann** [comment]
```
+ ssh ec2-user@3.133.90.102 'sudo /tmp/install.sh'
kex_exchange_identification: Connection closed by remote host
```

/test e2e-openshift-conformance-reduced

File: scripts/devenv-builder/configure-vm.sh [hunk 1]
```diff
@@ -88,6 +88,22 @@ if ${INSTALL_BUILD_DEPS} || ${BUILD_AND_RUN}; then
     sudo systemctl enable --now cockpit.socket
 fi
 
+GO_VER=1.20.4
+GO_ARCH=$([ "$(uname -i)" == "x86_64" ] && echo "amd64" || echo "arm64")
```
Comment (ggiguash) [inline scripts/devenv-builder/configure-vm.sh:92]:
```suggestion
GO_ARCH=$([ "$(uname -m)" == "x86_64" ] && echo "amd64" || echo "arm64")
```

File: scripts/devenv-builder/configure-vm.sh [hunk 2]
```diff
@@ -88,6 +88,22 @@ if ${INSTALL_BUILD_DEPS} || ${BUILD_AND_RUN}; then
...
+    # /usr/local/bin is before /usr/bin in $PATH so newer one is picked up
+    curl -L -o "go${GO_VER}.linux-${GO_ARCH}.tar.gz" "https://go.dev/dl/go${GO_VER}.linux-${GO_ARCH}.tar.gz" &&
+        sudo rm -rf "/usr/local/go${GO_VER}" && \
+            sudo mkdir -p "/usr/local/go${GO_VER}" && \
+            sudo tar -C "/usr/local/go${GO_VER}" -xzf "go${GO_VER}.linux-${GO_ARCH}.tar.gz" --strip-components 1 && \
+            sudo rm -rfv /usr/local/bin/{go,gofmt} && \
+            sudo ln --symbolic /usr/local/go${GO_VER}/bin/{go,gofmt} /usr/local/bin/ && \
+            rm -rfv "go${GO_VER}.linux-${GO_ARCH}.tar.gz"
+fi
+
```
Comment (ggiguash) [inline scripts/devenv-builder/configure-vm.sh:106]: We need to also add this to the devenv docs.

**Comment by dhellmann** [comment]
I think we're likely to want to take #1818 instead of this PR.

**Comment by dhellmann** [comment]
/close 

Let's take #1818 instead

**Pull request closed without merging**